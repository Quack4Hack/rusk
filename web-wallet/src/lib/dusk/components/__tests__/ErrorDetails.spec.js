import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { ErrorDetails } from "..";

describe("ErrorDetails", () => {
  const baseProps = {
    error: new Error("Some error messaage"),
    summary: "Some error summary",
  };

  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("should render the `ErrorDetails` component", () => {
    const { container } = render(ErrorDetails, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names to the rendered element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
    };
    const { container } = render(ErrorDetails, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render nothing if the error is `null`", () => {
    const props = {
      ...baseProps,
      error: null,
    };
    const { container } = render(ErrorDetails, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });
});
