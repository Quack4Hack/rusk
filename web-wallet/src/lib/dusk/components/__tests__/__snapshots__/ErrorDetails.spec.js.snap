// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorDetails > should pass additional class names to the rendered element 1`] = `
<details
  class="dusk-error-details dusk-error-details__details foo bar"
>
  <summary
    class="dusk-error-details__summary"
  >
    Some error summary
  </summary>
   
  <pre
    class="dusk-error-details__error"
  >
    			
    Some error messaage
    
		
  </pre>
</details>
`;

exports[`ErrorDetails > should render nothing if the error is \`null\` 1`] = ``;

exports[`ErrorDetails > should render the \`ErrorDetails\` component 1`] = `
<details
  class="dusk-error-details dusk-error-details__details"
>
  <summary
    class="dusk-error-details__summary"
  >
    Some error summary
  </summary>
   
  <pre
    class="dusk-error-details__error"
  >
    			
    Some error messaage
    
		
  </pre>
</details>
`;
